gevent (master branch) require cython to be present to execute setup.py

Hi,

```
$ pip install git+https://github.com/surfly/gevent.git#egg=gevent
Downloading/unpacking gevent from git+https://github.com/surfly/gevent.git
  Cloning https://github.com/surfly/gevent.git to ./build/gevent
  Running setup.py (path:/Users/mouadbenchchaoui/test/build/gevent/setup.py) egg_info for package gevent
    /Users/mouadbenchchaoui/test/bin/python util/cythonpp.py -o gevent.corecext.c gevent/core.ppyx
    Running cython -o gevent.corecext.c gevent/core.pyx  # !EV_USE_SIGNALFD && !defined(LIBEV_EMBED) && !defined(_WIN32)
    sh: cython: command not found
    Traceback (most recent call last):
      File "util/cythonpp.py", line 772, in <module>
        process_filename(filename, options.output_file)
      File "util/cythonpp.py", line 87, in process_filename
        output = run_cython(pyx_filename, sourcehash, output_filename, banner, comment)
      File "util/cythonpp.py", line 535, in run_cython
        system(command, comment)
      File "util/cythonpp.py", line 545, in system
        raise AssertionError('%r failed with code %s' % (command, result))
    AssertionError: 'cython -o gevent.corecext.c gevent/core.pyx' failed with code 32512
    make: *** [gevent/gevent.corecext.c] Error 1
    Running 'make' in /Users/mouadbenchchaoui/test/build/gevent
    Complete output from command python setup.py egg_info:
    /Users/mouadbenchchaoui/test/bin/python util/cythonpp.py -o gevent.corecext.c gevent/core.ppyx

Running cython -o gevent.corecext.c gevent/core.pyx  # !EV_USE_SIGNALFD && !defined(LIBEV_EMBED) && !defined(_WIN32)

sh: cython: command not found

Traceback (most recent call last):

  File "util/cythonpp.py", line 772, in <module>

    process_filename(filename, options.output_file)

  File "util/cythonpp.py", line 87, in process_filename

    output = run_cython(pyx_filename, sourcehash, output_filename, banner, comment)

  File "util/cythonpp.py", line 535, in run_cython

    system(command, comment)

  File "util/cythonpp.py", line 545, in system

    raise AssertionError('%r failed with code %s' % (command, result))

AssertionError: 'cython -o gevent.corecext.c gevent/core.pyx' failed with code 32512

make: *** [gevent/gevent.corecext.c] Error 1

Running 'make' in /Users/mouadbenchchaoui/test/build/gevent
```

Now this problem happen because gevent run make for also a call like `python setup.py egg_info`, the fun part is even with a call like `python setup.py --help` the same error will show up :).

This is a known problem in pip (please refer to: https://github.com/pypa/pip/issues/25), other has worked around it (for example check https://github.com/Unidata/netcdf4-python/pull/257/files), setuptools have something for this **setup_requires**, distutils not so much. All of this are just solution that come to my mind, sadly I am not sure of the reason for the make call there, so sorry but I can't help any further.

--- a/setup.py
+++ b/setup.py
@@ -285,7 +285,14 @@ def read(name, *args):
         return ''
 
 
-if PYPY:
+# If we are running info / help commands, we don't need to build anything
+if len(sys.argv) >= 2 and ('--help' in sys.argv[1:] or
+   sys.argv[1] in ('--help-commands', 'egg_info', '--version', 'clean')):
+    ext_modules = []
+    install_requires = [] if PYPY else ['greenlet']
+    include_package_data = PYPY
+    run_make = False
+elif PYPY:
     sys.path.insert(0, '.')
     # XXX ugly - need to find a better way
     system('cp -r libev gevent/libev')
@@ -335,7 +342,8 @@ def run_setup(ext_modules, run_make):
             "Topic :: Internet",
             "Topic :: Software Development :: Libraries :: Python Modules",
             "Intended Audience :: Developers",
-            "Development Status :: 4 - Beta"])
+            "Development Status :: 4 - Beta"]
+    )
 
 
 if __name__ == '__main__':

